ClassNotFoundException: CamelCronService

Camel-quarkus is failing to start if camel-quarkus-quartz added to a project

```xml
<dependency>
    <groupId>org.apache.camel.quarkus</groupId>
    <artifactId>camel-quarkus-quartz</artifactId>
    <version>1.0.0-M7</version>
</dependency>
```

```
Exception in thread "main" java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.base/java.lang.Class.newInstance(Class.java:584)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:60)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:558)
        ... 9 more
Caused by: java.lang.NoClassDefFoundError: org/apache/camel/component/cron/api/CamelCronService
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:801)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:699)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:622)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:580)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:398)
        at io.quarkus.deployment.steps.Core$factoryFinderResolver29.deploy_0(Core$factoryFinderResolver29.zig:278)
        at io.quarkus.deployment.steps.Core$factoryFinderResolver29.deploy(Core$factoryFinderResolver29.zig:36)
        at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:501)
        ... 9 more
Caused by: java.lang.ClassNotFoundException: org.apache.camel.component.cron.api.CamelCronService
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:582)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 23 more

```

--- a/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/BuildProcessor.java
+++ b/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/BuildProcessor.java
@@ -164,7 +164,6 @@ class BuildProcessor {
                     "META-INF/services/org/apache/camel/configurer/*",
                     "META-INF/services/org/apache/camel/language/*",
                     "META-INF/services/org/apache/camel/dataformat/*",
-                    "META-INF/services/org/apache/camel/cron/*",
                     "META-INF/services/org/apache/camel/send-dynamic/*"));
 
             // TODO: this can be removed and the RestBindingJaxbDataFormatFactory can be discovered

--- a/extensions/cron/deployment/src/main/java/org/apache/camel/quarkus/component/cron/deployment/CronProcessor.java
+++ b/extensions/cron/deployment/src/main/java/org/apache/camel/quarkus/component/cron/deployment/CronProcessor.java
@@ -20,6 +20,7 @@ import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 import org.apache.camel.component.cron.api.CamelCronConfiguration;
+import org.apache.camel.quarkus.core.deployment.CamelServicePatternBuildItem;
 import org.apache.camel.quarkus.core.deployment.UnbannedReflectiveBuildItem;
 
 class CronProcessor {
@@ -39,5 +40,11 @@ class CronProcessor {
     @BuildStep
     ReflectiveClassBuildItem registerForReflection() {
         return new ReflectiveClassBuildItem(true, false, CamelCronConfiguration.class);
+    }
+
+    @BuildStep
+    CamelServicePatternBuildItem camelCronServicePattern() {
+        return new CamelServicePatternBuildItem(CamelServicePatternBuildItem.CamelServiceDestination.DISCOVERY, true,
+                "META-INF/services/org/apache/camel/cron/*");
     }
 }

